### DISCUSSION TITLE
Error in minibuffer-exit-hook (vertico-buffer--restore): (error "Attempt to delete minibuffer or sole ordinary window")

### DISCUSSION THREAD

**Discussion by hab25** [minad/vertico#496] [Q&A] [answered]
Environment: emacs 29.3, vertico 1.8, consult 1.7

Reproduce with `emacs -Q`:

1. evaluate
```
(progn
  (require 'consult)
  (require 'vertico-buffer)
  (require 'vertico-multiform)
  (setopt vertico-multiform-commands
   	  `((,#'consult-ripgrep
   	     .
   	     (buffer
	      ;; works nicely with consult preview when `read-minibuffer-restore-windows` is `t` (the default)
   	      (vertico-buffer-display-action ,#'display-buffer-full-frame)))))
  (vertico-mode)
  (vertico-multiform-mode)
  (select-window (display-buffer-in-side-window (get-buffer "*Messages*") nil)))
```

2. type `M-x` -> "consult-ripgrep" -> `RET` (make sure the `*Messages*` buffer is selected in the side window, which should be the state the code ended at)
3. quit out immediately (i.e. without typing anything else) by typing `ESC ESC ESC`
4. note the message `Error in minibuffer-exit-hook (vertico-buffer--restore): (error "Attempt to delete minibuffer or sole ordinary window")` in the `*Messages*` buffer

Optionally,

5. repeat step 2
6. type a string that does match something your home directory, perhaps "bash", and then quit by doing `ESC ESC ESC`
7. observe that no new (i.e. from step 5 onwards) error message appeared in the `*Messages*` buffer

**Comment by minad** [comment]
Hi, I have difficulties reproducing this problem on Emacs 30. Can you please check if the problem is still present there? (FWIW I consider vertico-buffer-display-action=display-buffer-full-frame with consult-ripgrep a misconfiguration, since the Consult preview will split the frame anyway).

**Reply by hab25** [reply]
>  Can you please check if the problem is still present there?

Reproduced, emacs 30.0.50, the exact commit is 6c85b0311127332621e79217e7a208b679c89d5a. I can provide the compilation flags if necessary, they have been customized.

Here's a screencast showing it.

[Screencast from 24-06-24 22:12:23.webm](https://github.com/minad/vertico/assets/31970961/4cbc3d8e-9478-4948-974c-5534842f483f)

> FWIW I consider vertico-buffer-display-action=display-buffer-full-frame with consult-ripgrep a misconfiguration, since the Consult preview will split the frame anyway

The fact that preview will split the frame is desired. My intent with such a setting is to make sure, no matter what the current window configuration is, that these two windows will be the only ones visible while the consult command is running. In my real configuration, I use `(setopt split-height-threshold nil)` and I find this resulting full-frame side-by-side display very nice.

If the previous paragraph has made my intent more clear, do you still consider it a misconfiguration? If so, could you please suggest a proper configuration that achieves this?

**Reply by minad** [reply]
That's odd. I don't see the error message when trying this. Maybe it would help to add an additional check here:

https://github.com/minad/vertico/blob/ba650a7ab90d66686ba787937ac9e71f749c598e/extensions/vertico-buffer.el#L183-L184

**Reply by hab25** [reply]
> That's odd. I don't see the error message when trying this. Maybe it would help to add an additional check here:
> 
> https://github.com/minad/vertico/blob/ba650a7ab90d66686ba787937ac9e71f749c598e/extensions/vertico-buffer.el#L183-L184

Given that nothing seems to be broken if one disregards the error message, that might be correct.

Just in case it helps with brainstorming the solution, note again that if one tries to initiate `consult-ripgrep` from a regular window (i.e. non-side), the error doesn't occur.

**Reply by minad** [reply]
> Just in case it helps with brainstorming the solution, note again that if one tries to initiate consult-ripgrep from a regular window (i.e. non-side), the error doesn't occur.

Ah, I probably missed that when reproducing it. Sorry I've over read that above.

**Reply by minad** [reply, marked as answer]
Let's go with the simple fix for now: https://github.com/minad/vertico/commit/0d2cde6c964833fe1f715921d9b174aee9b40b24

Please let me know if this still breaks in some circumstances.

**Reply by hab25** [reply]
> Let's go with the simple fix for now: [0d2cde6](https://github.com/minad/vertico/commit/0d2cde6c964833fe1f715921d9b174aee9b40b24)
> 
> Please let me know if this still breaks in some circumstances.

Thank you; I will.

**Reply by nerffan1** [reply]
> Let's go with the simple fix for now: [0d2cde6](https://github.com/minad/vertico/commit/0d2cde6c964833fe1f715921d9b174aee9b40b24)
> 
> Please let me know if this still breaks in some circumstances.

Sorry to bother you, I am using doom emacs, so I am not sure if I should go to someone else or to this repo. I am using special characters with different encoding and I am unable to delete the vertico minibuffer when I try to choose a different type of encoding:
![image](https://github.com/user-attachments/assets/14c20393-c455-42a5-81a1-c3d8eebf34e5)
I am not able to delete this and I have to close all of emacs to get rid of this. Any ideas what I can do?
